lib/galaxy/datatypes/binary.py

```diff
@@ -4302,3 +4302,28 @@ def display_peek(self, dataset: DatasetProtocol) -> str:
             return dataset.peek
         except Exception:
             return "Binary Numpy npz file (%s)" % (nice_size(dataset.get_size()))
+
+
+@build_sniff_from_prefix
+class FITS(Binary):
+    """
+    FITS (Flexible Image Transport System) file data format, widely used in astronomy
+    Represents sky images (in celestial coordinates) and tables 
+     https://fits.gsfc.nasa.gov/fits_primer.html
+    """
+
+    file_ext = "fits"
+
+    def sniff_prefix(self, file_prefix: FilePrefix):
```

Comment:
Can you try to be more specific in the sniffing. In order to avoid false positives.

Reply:
Do this sort of records happen often?

That's what [libmagic](https://github.com/file/file/blob/master/magic/Magdir/images#L1335) uses to recognize this format, so it's somewhat widespread.

If it's really likely to cause misidentification, I could try to add extra details, but it might miss some cases, and will not be the same as libmagic (`file` command).

Reply:
Thanks for the info. In this case it should be fine :)